br/pkg/lightning/web/progress.go

```diff
@@ -108,11 +109,25 @@ type taskProgress struct {
 	checkpoints checkpointsMap
 }
 
-var currentProgress = taskProgress{
-	checkpoints: makeCheckpointsMap(),
+var (
+	currentProgress *taskProgress
+	// whether progress is enabled
+	progressEnabled = atomic.NewBool(false)
+)
+
+// EnableCurrentProgress init current progress struct on demand.
+// NOTE: this call is not thread safe, so it should only be inited once at the very beginning of progress start.
+func EnableCurrentProgress() {
+	currentProgress = &taskProgress{
+		checkpoints: makeCheckpointsMap(),
+	}
+	progressEnabled.Store(true)
```

Comment:
How about using `progressEnabled.CAS(false, ture)` to avoid data race?

Reply:
Do you mean call `CAS` and init `currentProgress` if cas returns false? Since `InitCurrentProgress` should only be called once, this seems unnecessary.

Reply:
Yes. But this is not important. You can just ignore it.